File: TheLionsDen.Model/Responses/RoomResponse.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace TheLionsDen.Model.Responses
{
    public class RoomResponse
    {
        public int RoomId { get; set; }
        public string Name { get; set; } 
        public float Price { get; set; }
        public int? Number { get; set; }
        public int? Floor { get; set; }
        public string State { get; set; } 
        public int RoomTypeId { get; set; }

        public string Amenities { get; set; } = "No amenities assigned.";
        public string RoomTypeName { get; set; }
        [JsonIgnore]
        public virtual RoomTypeResponse RoomType { get; set; } 

        //public virtual ICollection<Favourite> Favourites { get; set; }
        //public virtual ICollection<Reservation> Reservations { get; set; }
        [JsonIgnore]
        public virtual ICollection<RoomAmenityResponse> RoomAmenities { get; set; }
    }
}

File: the_lions_den_mobile/lib/model/room/room_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:the_lions_den_mobile/model/amenity/amenity_response.dart';
import 'package:the_lions_den_mobile/model/room_amenity/room_amenity_response.dart';
import 'package:the_lions_den_mobile/model/room_type/room_type_response.dart';
part 'room_response.g.dart';

@JsonSerializable()
class RoomResponse {
  int? roomId;
  double? price;
  String? name;
  String? amenities;
  String? roomTypeName;
  String? coverImage;
  RoomTypeResponse? roomType;
  List<RoomAmenityResponse>? roomAmenities;

  RoomResponse();

  factory RoomResponse.fromJson(Map<String, dynamic> json) =>
      _$RoomResponseFromJson(json);

  /// Connect the generated [_$RoomResponseToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$RoomResponseToJson(this);
}
